CMakeLists.txt

```diff
@@ -5,7 +5,10 @@
 cmake_minimum_required(VERSION 3.1)
 project(TYPE_SAFE)
 
-include(external/external.cmake)
+find_package(debug_assert)
+if(NOT debug_assert_FOUND)
+    include(external/external.cmake)
```

Comment:
So 3.1 is not supported due to using `target_sources` on an interface library, which is unrelated to this PR. The cmake version should probably update to 3.3 then.